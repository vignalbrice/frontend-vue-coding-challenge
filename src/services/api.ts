import ky from 'ky';

export const api = ky.create({
  prefixUrl: `${window.location.origin}/api`,
});
