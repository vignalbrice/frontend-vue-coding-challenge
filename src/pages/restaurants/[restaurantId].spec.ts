import { flushPromises, mount } from '@vue/test-utils';

import PageRestaurant from './[restaurantId].vue';
import restaurantsResponse from '@/mock/restaurants.json';
import * as apiService from '~/services/api';

const restaurant = restaurantsResponse[0];

vi.spyOn(apiService, `api`)
  .mockImplementationOnce(() => ({ json: vi.fn().mockResolvedValue(restaurant) } as any))
  .mockImplementationOnce(() => ({ json: vi.fn().mockRejectedValueOnce(new Error(`something went wrong`)) } as any));

describe(`page-restaurant`, () => {
  beforeEach(async (context) => {
    context.router.push(`/restaurants/${restaurant.id}`);
    await context.router.isReady();
  });

  it(`displays the list of restaurants`, async (context) => {
    const wrapper = mount(PageRestaurant, {
      global: { plugins: [context.router] },
    });
    await flushPromises();
    expect(apiService.api).toHaveBeenCalled();
    expect(apiService.api).toHaveBeenCalledWith(`restaurants/${restaurant.id}`);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it(`handles fetch error`, async (context) => {
    const wrapper = mount(PageRestaurant, {
      global: { plugins: [context.router] },
    });
    await flushPromises();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
