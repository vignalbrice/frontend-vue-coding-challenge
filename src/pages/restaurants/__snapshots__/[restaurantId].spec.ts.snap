// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`page-restaurant > displays the list of restaurants 1`] = `
"<div data-v-70ec466b=\\"\\" class=\\"restaurant-container grid grid-cols-[minmax(0,_1fr)_16rem] gap-6\\">
  <div data-v-70ec466b=\\"\\" class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
    <!---->
    <div class=\\"v-card__loader\\">
      <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
        <!---->
        <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <v-img-stub data-v-70ec466b=\\"\\" height=\\"250\\" gradient=\\"to top, rgba(0,0,0,.1), rgba(0,0,0,.5)\\" cover=\\"true\\" eager=\\"false\\" options=\\"[object Object]\\" src=\\"https://s3-media2.fl.yelpcdn.com/bphoto/RDKmsxZZczP4EiTCB9AYOw/o.jpg\\" inline=\\"false\\" transition=\\"fade-transition\\"></v-img-stub>
    <div data-v-70ec466b=\\"\\" class=\\"v-card-text\\">
      <div data-v-70ec466b=\\"\\" class=\\"grid grid-cols-2 gap-4\\">
        <div data-v-70ec466b=\\"\\" class=\\"flex gap-2\\"><i class=\\"mdi-map-marker mdi v-icon notranslate v-theme--light v-icon--size-default text-slate-300\\" aria-hidden=\\"true\\"></i>
          <div class=\\"text-slate-600\\"><pre class=\\"my-0\\">Jungfernstieg 16 - 20
4. OG
20354 Hamburg
Germany</pre>
          </div>
        </div>
        <div data-v-70ec466b=\\"\\" class=\\"flex gap-2\\"><i class=\\"mdi-phone mdi v-icon notranslate v-theme--light v-icon--size-default text-slate-300\\" aria-hidden=\\"true\\"></i>
          <div class=\\"text-slate-600\\">
            <p data-v-70ec466b=\\"\\" class=\\"text-body-1\\">+(494) 057-283299</p>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <!----><span class=\\"v-card__underlay\\"></span>
  </div>
  <aside data-v-70ec466b=\\"\\">
    <ul data-v-70ec466b=\\"\\" class=\\"pa-0\\">
      <li data-v-6426699a=\\"\\" data-v-70ec466b=\\"\\" class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated va-company-review\\">
        <!---->
        <div class=\\"v-card__loader\\">
          <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div data-v-6426699a=\\"\\" class=\\"d-flex justify-center mt-auto text-h5 ml-2\\"> Rating overview </div>
        <div data-v-6426699a=\\"\\" class=\\"va-rating d-flex align-center flex-column my-auto\\">
          <div data-v-6426699a=\\"\\" class=\\"va-rating-review\\">
            <h2 data-v-6426699a=\\"\\">4.0</h2>
            <h5 data-v-6426699a=\\"\\">/5</h5>
          </div>
          <div data-v-6426699a=\\"\\" class=\\"v-rating v-rating--hover v-theme--light\\"><label for=\\"v-rating-0-0\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 0 of 5</span>
              <!---->
            </label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-0\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"0\\">
            <div class=\\"v-rating__wrapper\\">
              <!---->
              <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-1\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 1 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-1\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"1\\"></div>
              <!---->
            </div>
            <div class=\\"v-rating__wrapper\\">
              <!---->
              <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-2\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 2 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-2\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"2\\"></div>
              <!---->
            </div>
            <div class=\\"v-rating__wrapper\\">
              <!---->
              <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-3\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 3 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-3\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"3\\"></div>
              <!---->
            </div>
            <div class=\\"v-rating__wrapper\\">
              <!---->
              <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-4\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 4 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-4\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"4\\"></div>
              <!---->
            </div>
            <div class=\\"v-rating__wrapper\\">
              <!---->
              <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-5\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #333; caret-color: #333; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-5\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"5\\"></div>
              <!---->
            </div>
          </div>
          <div data-v-6426699a=\\"\\" class=\\"px-3\\">3 ratings </div>
        </div>
        <div data-v-6426699a=\\"\\" class=\\"px-3\\"> Reviews </div>
        <div data-v-6426699a=\\"\\" class=\\"v-card-text\\">
          <div data-v-6426699a=\\"\\" class=\\"v-list v-theme--light bg-transparent v-list--density-compact v-list--one-line d-flex flex-column-reverse\\" tabindex=\\"0\\" role=\\"listbox\\">
            <div data-v-6426699a=\\"\\" class=\\"v-list-item v-theme--light v-list-item--density-compact v-list-item--one-line v-list-item--variant-text\\">
              <!----><span class=\\"v-list-item__underlay\\"></span>
              <!---->
              <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
                <!---->
                <!---->
                <!---->
              </div>
              <div class=\\"v-list-item__append\\">
                <div data-v-6426699a=\\"\\" class=\\"rating-values\\"><span data-v-6426699a=\\"\\" class=\\"d-flex justify-end ml-2 review-text\\">I've had decent lunches at Billy The Butcher a couple of times now. It's what I would describe as a trendy burger joint on the 4th floor of the Alsterhaus...</span></div>
              </div>
            </div>
            <div data-v-6426699a=\\"\\" class=\\"v-list-item v-theme--light v-list-item--density-compact v-list-item--one-line v-list-item--variant-text\\">
              <!----><span class=\\"v-list-item__underlay\\"></span>
              <!---->
              <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
                <!---->
                <!---->
                <!---->
              </div>
              <div class=\\"v-list-item__append\\">
                <div data-v-6426699a=\\"\\" class=\\"rating-values\\"><span data-v-6426699a=\\"\\" class=\\"d-flex justify-end ml-2 review-text\\">We arrived Hamburg at noon and decide to get some burgers before going to the office. This place is pretty close to the city hall and rated well.

I ordered...</span></div>
              </div>
            </div>
            <div data-v-6426699a=\\"\\" class=\\"v-list-item v-theme--light v-list-item--density-compact v-list-item--one-line v-list-item--variant-text\\">
              <!----><span class=\\"v-list-item__underlay\\"></span>
              <!---->
              <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
                <!---->
                <!---->
                <!---->
              </div>
              <div class=\\"v-list-item__append\\">
                <div data-v-6426699a=\\"\\" class=\\"rating-values\\"><span data-v-6426699a=\\"\\" class=\\"d-flex justify-end ml-2 review-text\\">Nice place to have good burger for lunch. Most of personnel speak English which is good. Prices are decent, also I would recommend to share fries with a...</span></div>
              </div>
            </div>
          </div>
        </div>
        <!---->
        <!----><span class=\\"v-card__underlay\\"></span>
      </li>
    </ul>
  </aside>
</div>"
`;

exports[`page-restaurant > handles fetch error 1`] = `
"<div data-v-70ec466b=\\"\\" class=\\"v-alert v-theme--light text-warning v-alert--density-default v-alert--variant-tonal mt-4\\" role=\\"alert\\">
  <!----><span class=\\"v-alert__underlay\\"></span>
  <!---->
  <div class=\\"v-alert__prepend\\"><i class=\\"mdi-alert-circle mdi v-icon notranslate v-theme--light\\" style=\\"font-size: 28px; height: 28px; width: 28px;\\" aria-hidden=\\"true\\" density=\\"default\\"></i></div>
  <div class=\\"v-alert__content\\">
    <!---->warningsomething went wrong <br data-v-70ec466b=\\"\\">
  </div>
  <!---->
  <!---->
</div>"
`;
